dma: stm32: enable half transfer interrupt

shlomow opened the pull request:
Enable half transfer interrupt of the dma controller
in case the channel was enabled in circular mode.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>

I am not sure if this is the correct way to do this. Any opinions?
This mode can be quite useful when trying to continuously sampling
with adc controller.

shlomow commented:
@erwango

erwango reviewed (commented):
Seems fine a priori but would be nice to have example of usage.
A related PR that makes use of it for instance ?

FRASTM commented on drivers/dma/dma_stm32.c:
```diff
@@ -463,7 +463,11 @@ DMA_STM32_EXPORT_API int dma_stm32_configure(const struct device *dev,
 	LL_DMA_EnableIT_TC(dma, dma_stm32_id_to_stream(id));
-	/* Half-Transfer irq is not handled */
+
+	/* Enable Half-Transfer irq is circular mode is enabled */
```
/* Enable Half-Transfer irq **if** circular mode is enable */

shlomow committed:
dma: stm32: enable half transfer interrupt

Enable half transfer interrupt of the dma controller
in case the channel was enabled in circular mode.

Signed-off-by: Shlomi Vaknin <shlomi.39sd@gmail.com>

shlomow commented:
@erwango I am using half-transfer irq in #33846. I guess it will take a lot of time to this to be merged, but I it is
the best usage I could find for this. I hope it is enough.

shlomow commented:
@erwango Can you approve this?
